cmd/dex/config.go

```diff
@@ -181,6 +181,14 @@ var storages = map[string]func() StorageConfig{
 	"mysql":      func() StorageConfig { return new(sql.MySQL) },
 }
 
+// isExpandEnvEnabled returns if os.ExpandEnv should be used for each storage and connector config.
+// Disabling this feature avoids surprises e.g. if the LDAP bind password contains a dollar character.
+// True by default, to avoid a breaking change.
+// False if the env variable "DEX_EXPAND_ENV" is set to "0".
+func isExpandEnvEnabled() bool {
+	return os.Getenv("DEX_EXPAND_ENV") != "0"
```

Comment:
How about using [ParseBool](https://golang.org/pkg/strconv/#ParseBool) instead? Then a whole bunch of other boolean-like values can be accepted. `false/true` comes to mind as better alternatives to 0/1 and I'd also use those in the documentation instead of 0/1.

Obviously this means that you have to look up the env first.

Also, thinking about the variable name: since this is true by default, I wonder if  it would make sense to use `DEX_NO_EXPAND_ENV` instead.

Reply:
> How about using ParseBool instead?

Sounds good, will do that.

> I wonder if it would make sense to use DEX_NO_EXPAND_ENV instead.

I'm not a big fan of this, for the following reasons:
* Consider `DEX_NO_EXPAND_ENV=false`, which is a double negation. I wanted to avoid that case.
* In case that we later change the default to "don't expand", we would have to change back the variable name `DEX_NO_EXPAND_ENV` -> `DEX_EXPAND_ENV`.

E.g. the [OpenLDAP Docker image](https://github.com/osixia/docker-openldap) also uses `LDAP_...=false` instead of `LDAP_NO_...=true`
But if you insist, I can still change it.